src/main/java/toast/specialMobs/MobHelper.java

```diff
@@ -28,42 +26,37 @@
 import toast.specialMobs.entity.pigzombie.Entity_SpecialPigZombie;
 import toast.specialMobs.network.MessageExplosion;
 
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Objects;
+import java.util.stream.Stream;
+
 public abstract class MobHelper
 {
+    private static Stream<EntityAITaskEntry> typeFilteredTaskEntries(EntityLiving entity, Class<?> type) {
+        return Arrays.stream(entity.tasks.taskEntries.toArray())
+            .filter(Objects::nonNull)
+            .filter(e -> e instanceof EntityAITaskEntry)
```

Comment:
Pretty sure this list will only ever contain EntityAITaskEntry.

Reply:
The crashes and log spam indicates otherwise, ClassCastExceptions occur on exactly this list, trying to cast to EntityAITaskEntry

Reply:
It's a Java stupidness. If you don't call `toArray(new EntityAITaskEntry[0])` it will create a Object[]. Generic type information is not preserved.

Reply:
Should the code I just pushed avoid the exception then?

Reply:
Yeah it should be fine now.